src/app/core/entity/database-indexing/database-indexing.service.ts

```diff
@@ -73,10 +79,39 @@ export class DatabaseIndexingService {
 
   /**
    * Load data from the Database through the given, previously created index.
+   * @param entityConstructor
    * @param indexName The name of the previously created index to be queried.
-   * @param options Additional query options (see @link{Database})
+   * @param options (Optional) additional query options object or a simple value used as the exact key to retrieve
    */
-  queryIndex(indexName: string, options?: any) {
-    return this.db.query(indexName, options);
+  async queryIndexDocs<T extends Entity>(
+    entityConstructor: EntityConstructor<T>,
+    indexName: string,
+    options: QueryOptions | string = {}
+  ): Promise<T[]> {
+    if (typeof options === "string") {
+      options = { key: options };
+    }
+    options.include_docs = true;
+
+    const rawResults = await this.queryIndexRaw(indexName, options);
+    return rawResults.rows.map((loadedRecord) => {
+      const entity = new entityConstructor("");
+      this.entitySchemaService.loadDataIntoEntity(entity, loadedRecord.doc);
+      return entity;
+    });
+  }
+
+  async queryIndexStats(
+    indexName: string,
+    options: QueryOptions = {
+      reduce: true,
+      group: true,
+    }
+  ): Promise<any> {
+    return await this.queryIndexRaw(indexName, options);
+  }
+
+  async queryIndexRaw(indexName: string, options: QueryOptions): Promise<any> {
```

Comment:
With the changes in `search.compontent.ts`, this function can be made private
```suggestion
  private async queryIndexRaw(indexName: string, options: QueryOptions): Promise<any> {
```